### PR TITLE
Make OptProf MicroBuild plug-in work again for Official Builds

### PR DESCRIPTION
**Pull request by donnie-msft**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2749
 
Regression? Last working version:

## Description

Follow-up to https://github.com/NuGet/NuGet.Client/pull/5706

Copies pipeline parameter to a template variable in the parent YAML so that `Conditions` work. Most relevantly, this means OptProf MicroBuild plugin will be invoked by the 1ES templates.

Also remove outputs with conditions would never be true (ie, values never satisfy conditions for BuildRTM and IsOfficialBuild).

[Sample official build](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9306988&view=logs&j=d80b2849-30d4-52dd-74cd-d6536ba98fe0&t=834cc09b-7de8-53f7-7930-d1ac3e92b356)

[Sample OptProf](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9306989&view=results)
[Sample OptProf 2](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9307253&view=results)

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: eng/pipelines/templates/pipeline.yml [hunk 1]
```diff
@@ -106,39 +107,39 @@ stages:
         swix:
           enabled: true
         optprof:
-          enabled: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
```
Comment (nkolev92) [inline eng/pipelines/templates/pipeline.yml:109]: Why is the BuildRTM unnecessary? 
Wouldn't it now run  in 2 phases while it previous ran in 1?

Comment (nkolev92) [reply eng/pipelines/templates/pipeline.yml:109]: line: 95 hardcodes it.

**Review by nkolev92 — DISMISSED** [review]

File: eng/pipelines/templates/pipeline.yml (hunk 1 above)
Comment (zivkan) [reply eng/pipelines/templates/pipeline.yml:109]: Line 95 sets up a job variable, but it doesn't prevent this optprof step to skip on official build's RTM job.

File: eng/pipelines/templates/pipeline.yml [hunk 2]
```diff
@@ -106,39 +108,39 @@ stages:
...
+          enabled: ${{ variables['isOfficialBuild'] }}
```
Comment (zivkan) [inline eng/pipelines/templates/pipeline.yml:111]: you probably don't want to remove `succeeded()`, as there's no point installing the optprof plugin if a previous step already failed.

**Comment by donnie-msft** [comment]
> Line 95 sets up a job variable, but it doesn't prevent this optprof step to skip on official build's RTM job.

@zivkan could you rephrase this, please? Optprof only runs on NonRTM

File: eng/pipelines/templates/pipeline.yml (hunk 2 above)
Comment (donnie-msft) [reply eng/pipelines/templates/pipeline.yml:111]: Does this look right to you? I can't find documentation that this is supported within a variable. I also didn't see that the templates pass in a condition - but this is "compiling" at least:
https://github.com/NuGet/NuGet.Client/pull/5716/commits/81376bd6c90cf5206e1a542fa443617cb438f9fa#diff-ff6085ab5cf199ecbf9b59ba9a5854fdc078bdc1c7516d16c95f6bff5ecd070aR112

Comment (jeffkl) [reply eng/pipelines/templates/pipeline.yml:111]: The plug-ins are installed nearly first, after clone but before any of our tasks run.  I don't think we need a condition on this step.

File: eng/pipelines/templates/pipeline.yml (hunk 1 above)
Comment (zivkan) [reply eng/pipelines/templates/pipeline.yml:109]: > @zivkan could you rephrase this, please? Optprof only runs on NonRTM

Github is showing me the following diff on the files tab, but for some reason not on the conversation tab:

```diff
-enabled: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
+enabled: ${{ variables['isOfficialBuild'] }}
```

So previously it would only run on the non-RTM job, and only on official builds.

Now, it looks like it's running on official builds for any job.

However, while gathering evidence, I now realize this is part of the nonRTM job defined on line 87, whereas the RTM job definition starts on 271.  When I originally split up the jobs into different files, I intended the individual yamls to contain "all" the steps, including publishing, so now that pipeline.yml has all the outputs, it's completely thrown off my mental model for how the CI jobs are defined.

**Review by zivkan — DISMISSED** [review]

File: eng/pipelines/templates/pipeline.yml (hunk 2 above)
Comment (donnie-msft) [reply eng/pipelines/templates/pipeline.yml:111]: Discussed offline. No condition needed.

**Review by zivkan — APPROVED** [review]
